components/scream/src/physics/shoc/atmosphere_macrophysics.hpp

```diff
@@ -88,6 +89,33 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
       const Real inv_ggr = 1/ggr;
 
       const int nlev_packs = ekat::npack<Spack>(nlev);
+
+
+      /*---------------------------------------------------------------------------------
+       *Wet to dry mixing ratios:
+       *-------------------------
+       *Since tracers from the host model (or AD) are  wet mixing ratios and SHOC expects
+       *these tracers in dry mixing ratios, we convert the wet mixing ratios to dry mixing
+       *ratios for all the tracers *except* for qv [which is done after all the other tracers]
+       *and TKE [which is always defined in units of J/(kg of total air) and therefore
+       *should be considered to be "wet" when stored in the FM and also "wet" when used in
+       *SHOC].
+       *----------------------------------------------------------------------------------
+       */
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, num_qtracers-2), [&] (const Int& idx) {
+          int iq   = idx / nlev_packs;
```

Comment:
doesn't this loop only go for `idx` in `1...num_qtracers-2`?  If so, I don't see how we can grab `ilev` from `idx`.  Instead it would seem like `idx` is `iq`.  Am I missing something here?

Reply:
This comment still stands

Reply:
@AaronDonahue : Thanks for catching this. I added an additional kokkos parallel-for to loop over levels.